ui/src/components/WMWorkerPoolEditor/index.jsx

```diff
@@ -0,0 +1,326 @@
+            <TextField
+              label="Owner's Email"
+              name="owner"
+              error={!workerPool.owner.includes('@')}
+              onChange={this.handleInputChange}
+              fullWidth
+              value={workerPool.owner}
+              margin="normal"
+              required
+              helperText={
+                !workerPool.owner.includes('@') && 'Should be valid email'
+              }
+            />
+          </ListItem>
+
+          <ListItem>
+            <FormControlLabel
+              control={
+                <Switch
+                  checked={workerPool.emailOnError}
+                  onChange={this.handleSwitchChange}
+                  value="emailOnError"
+                />
+              }
+              label="Email the owner about errors"
+            />
+          </ListItem>
+        </List>
+
+        <List className={classes.middleList}>
+          <ListSubheader>Provider</ListSubheader>
+          <ListItem>
+            <TextField
+              id="select-provider-type"
+              className={classes.dropdown}
+              select
+              helperText="Which cloud do you want to run your tasks in?"
+              value={providerType}
+              name="providerType"
+              onChange={this.handleProviderTypeChange}
+              margin="normal">
+              {Array.from(PROVIDERS.keys()).map(p => (
+                <MenuItem key={p} value={p}>
+                  {p}
+                </MenuItem>
+              ))}
+            </TextField>
+          </ListItem>
+        </List>
+
+        <List className={classes.list}>
+          <ListSubheader>Configuration:</ListSubheader>
+          <ListItem>
+            <CodeEditor
+              value={JSON.stringify(workerPool.config, null, 2)}
+              onChange={this.handleEditorChange}
+              lint
+            />
+          </ListItem>
+
+          <Button
+            spanProps={{
+              className: allowEditWorkerPoolId
+                ? classes.createIconSpan
+                : classes.saveIconSpan,
+            }}
+            name="saveRequest"
+            disabled={invalidProviderConfig || actionLoading}
+            requiresAuth
+            tooltipProps={{ title: 'Save Worker Pool' }}
+            onClick={this.handleOnClick}
+            classes={{ root: classes.saveIcon }}
+            variant="round">
+            <ContentSaveIcon />
+          </Button>
+
+          {!allowEditWorkerPoolId && (
+            <SpeedDial>
+              <SpeedDialAction
+                name="deleteRequest"
+                requiresAuth
+                tooltipOpen
+                icon={<DeleteIcon />}
+                onClick={this.handleOnClick}
+                tooltipTitle="Delete"
+                className={classes.deleteIcon}
+                disabled={actionLoading}
+              />
+            </SpeedDial>
+          )}
+        </List>
+      </Fragment>
+    );
+  }
+}
```

Comment:
STR:

1. Navigate to http://localhost:5080/worker-manager/create

Actual:

![Screen Shot 2019-07-03 at 10 15 00 AM](https://user-images.githubusercontent.com/3766511/60598936-9651eb00-9d7b-11e9-9a2a-06e6a1813f8c.png)


Expected: Initially the error state is not showing

Apologies for not catching this earlier.

Reply:
This actually was intentional to mark the required fields

Reply:
Required fields are shown via the "*" beside the label. I don't think there is a need to show an error on page load.